### PR TITLE
Add eslint infrastructure

### PR DESCRIPTION
**Pull request by loay**

### CODE REVIEW

**Comment by loay** [comment]
@superkhau PTAL Thanks

**Comment by Amir-61** [comment]
@loay did you see this failure:
```
/mnt/jenkins/workspace/loopback-connector-postgresql/89d66e82/test/postgresql.test.js:52
getDS = function(config) {
      ^
ReferenceError: getDS is not defined
```

File: package.json [hunk 1]
```diff
@@ -14,19 +14,21 @@
...
-    "async": "^0.9.0",
-    "debug": "^2.1.1",
-    "loopback-connector": "^2.1.0",
-    "pg": "^6.0.0",
-    "strong-globalize": "^2.6.2"
+    "async": "^0.9.2",
+    "debug": "^2.2.0",
+    "loopback-connector": "^2.4.0",
+    "pg": "^6.0.3",
+    "strong-globalize": "^2.6.6"
```
Comment (superkhau) [inline package.json]: You should be upgrading these in a separate PR. This has nothing to do with linting.

File: package.json [hunk 2]
```diff
@@ -14,19 +14,21 @@
...
-    "should": "^8.0.2",
-    "sinon": "^1.15.4"
+    "bluebird": "^3.4.1",
+    "eslint": "^2.13.1",
+    "eslint-config-loopback": "^4.0.0",
+    "loopback-datasource-juggler": "^2.47.0",
+    "mocha": "^3.0.2",
+    "rc": "^1.1.6",
+    "should": "^8.4.0",
+    "sinon": "^1.17.5"
```
Comment (superkhau) [inline package.json]: ditto

**Comment by superkhau** [comment]
@loay Do not upgrade the dependencies for a linting PR. Please remove those changes in `package.json`, then LGTM to merge the rest. Ping me again in the other PR for upgrading dependencies. Please look into the CI failures too and make sure they are unrelated to this PR.

**Comment by superkhau** [comment]
While you're at it, make sure you fix the issue in @Amir-61's comment above regarding getDS breaking things before you merge.

**Comment by rmg** [comment]
@loay sorry, I merged #158 and now you have conflicts to resolve :-(

**Comment by loay** [comment]
@rmg No worries, thanks for letting me know. I am rebasing now :)

File: package.json [hunk 3]
```diff
@@ -11,7 +11,10 @@
   ],
   "main": "index.js",
   "scripts": {
-    "test": "mocha -R spec --timeout 10000 --require test/init.js test/*.test.js"
+    "pretest": "node pretest.js",
+    "lint": "eslint .",
+    "test": "mocha --timeout 10000 test/*.js",
```
Comment (rmg) [inline package.json]: I think you'll need to restore the `--require test/init.js` because of how the `getDataSource` and `getDBConfig` globals are used in the tests.

File: package.json (hunk 1 above)
Comment (loay) [inline package.json]: Fixed

**Comment by loay** [comment]
@superkhau @jannyHou PTAL. Thanks.

**Comment by jannyHou** [comment]
@loay LGTM 👍

**Comment by superkhau** [comment]
LGTM
